bson/src/main/org/bson/io/ByteBufferBsonInput.java

```diff
@@ -152,27 +149,25 @@ private String readString(final int size) {
             return ONE_BYTE_ASCII_STRINGS[asciiByte];  // this will throw if asciiByte is negative
         } else {
             byte[] bytes = new byte[size - 1];
-            readBytes(bytes);
-            byte nullByte = readByte();
+            buffer.get(bytes);
+            byte nullByte = buffer.get();
             if (nullByte != 0) {
                 throw new BsonSerializationException("Found a BSON string that is not null-terminated");
             }
             return new String(bytes, UTF8_CHARSET);
         }
     }
 
-    private void readUntilNullByte() {
-        //CHECKSTYLE:OFF
-        while (readByte() != 0) { //NOPMD
-            //do nothing - checkstyle & PMD hate this, not surprisingly
-        }
-        //CHECKSTYLE:ON
-    }
-
     @Override
     public void skipCString() {
         ensureOpen();
-        readUntilNullByte();
+        boolean checkNext = true;
+        while (checkNext) {
+            if (!buffer.hasRemaining()) {
```

Comment:
Can we get rid of this check and just catch an ArrayIndexOutOfBoundsException from `buffer.get()`?

Reply:
try / catch is much slower

Reply:
Since the benchmarks show an improvement, I'll resolve this.